### ISSUE TITLE
Registry authentication problem while empty 'auth_token' value is returned by catalog

### ISSUE THREAD

**Issue by vifrrg** [davedoesdev/dxf#27]
Greetings.
I've faced the problem of registry authentication when response for token request contains the key 'auth_token', but its value is empty. Some harbor registries doing so. Example:
`{
  "token": "_long_but_correct_blahblahblah_token_so_I_cut_it_off_",
  "access_token": "",
  "expires_in": 1800,
  "issued_at": "2019-11-27T08:50:25Z"
}`

The suggested fix is in __init.py__ file, please see here:
https://github.com/vifrrg/dxf/commit/2d82a0adf6cd0ad73db0dd0700b1f203dddc7a36
Thank You.

**Comment by davedoesdev** [comment]
@vifrrg thanks for letting me know, I see the issue. I'll integrate the fix.

**Comment by davedoesdev** [comment]
@vifrrg I don't think we need to check `len(rjson['access_token'])` if we've already checked `rjson['access_token']`.

**Comment by davedoesdev** [comment]
I was thinking of:

```diff
diff --git a/dxf/__init__.py b/dxf/__init__.py
index 9b48aae..304daa0 100644
--- a/dxf/__init__.py
+++ b/dxf/__init__.py
@@ -305,7 +305,8 @@ class DXFBase(object):
                 r = self._sessions[0].get(auth_url, headers=headers, verify=self._tlsverify)
             _raise_for_status(r)
             rjson = r.json()
-            self.token = rjson['access_token'] if 'access_token' in rjson else rjson['token']
+            # Use 'access_token' value if present and not empty, else 'token' value.
+            self.token = rjson.get('access_token') or rjson['token']
             return self._token
 
         self._headers = headers
```

**Comment by vifrrg** [comment]
@davedoesdev
1. As to checking length: it was my bad habbit from one of previous python version where empty string was treated as 'True' in 'if'. So the code like `if(""): print("blahblahblah")` was printing that "blahblahblah". This behaviour was changed long time ago and I'm still forgetting it every time. :)
In recent python interpreters it is not really needed. You right.

2. As to Your fix suggestion:
Yes, it looks more compact and readable then mine. Let me check it in my environment. If it works by the right way - let`s integrate it.
I'll let You know the result in a few hours.

**Comment by vifrrg** [comment]
@davedoesdev, Your fix works fine in both cases (old registry without "auth_token" and new one with empty value in it)
May be applied to next release.
Thank You.

**Comment by davedoesdev** [comment]
@vifrrg released in version 7.3.5
Appreciate the fix!
